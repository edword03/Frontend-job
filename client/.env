REACT_APP_API_URL_CITY='https://api.hh.ru/areas'
REACT_APP_V=How to
REACT_APP_API_URL_GET_VACANCIES='https://api.hh.ru/vacancies'