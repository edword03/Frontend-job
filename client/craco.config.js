// const path = require('path');

// module.exports = {
//   resolve: {
//     extensions: ['js', 'ts'],
//     alias: {
//       '@': path.resolve(__dirname, 'src'),
//       '@assets': path.resolve(__dirname, 'src/components'),
//       '@components': path.resolve(__dirname, 'src/components'),
//       '@styles': path.resolve(__dirname, 'src/styles'),
//       '@utils': path.resolve(__dirname, 'src/utils'),
//       // ...etc
//     },
//   },
// }

const path = require('path');

const resolvePath = p => path.resolve(__dirname, p)

module.exports = {
    // ...
    webpack: {
        alias: {
            '@': resolvePath('./src'),
            '@assets': resolvePath('./src/assets'),
            '@components': resolvePath('./src/components'),
            '@styles': resolvePath('./src/styles'),
            '@types': resolvePath('./src/types'),
            "@utils": resolvePath('./src/utils'),
            '@pages': resolvePath('./src/pages'),
            "@services": resolvePath('./src/services')
        }
    },
  // ...
}