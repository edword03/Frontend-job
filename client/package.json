{
  "name": "frontend-job",
  "version": "0.1.0",
  "private": true,
  "engines": {
    "node": "16.x"
  },
  "dependencies": {
    "@apollo/client": "^3.3.21",
    "@craco/craco": "^6.4.3",
    "@testing-library/jest-dom": "^5.14.1",
    "@testing-library/react": "^11.2.7",
    "@testing-library/user-event": "^12.8.3",
    "@types/jest": "^26.0.24",
    "@types/node": "^12.20.16",
    "@types/react": "^17.0.14",
    "@types/react-dom": "^17.0.9",
    "@types/styled-components": "^5.1.19",
    "graphql": "^15.5.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-responsive": "^9.0.0-beta.5",
    "react-scripts": "4.0.3",
    "styled-components": "^5.3.3",
    "typescript": "^4.3.5",
    "typescript-plugin-css-modules": "^3.4.0",
    "web-vitals": "^1.1.2"
  },
  "scripts": {
    "start": "craco start",
    "build": "craco build",
    "test": "craco test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "classnames": "^2.3.1"
  }
}
