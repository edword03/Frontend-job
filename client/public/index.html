<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <link rel="icon" href="favicon.ico" type="image/x-icon" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="theme-color" content="#000000" />
    <meta
      name="description"
      content="Сервис по поиску вакансий web-разработки"
    />
    <link rel="apple-touch-icon" href="touch-icon-iphone.png?ver=e3394" />
    <link
      rel="apple-touch-icon"
      sizes="76x76"
      href="touch-icon-ipad.png?ver=e3394"
    />
    <link
      rel="apple-touch-icon"
      sizes="120x120"
      href="touch-icon-iphone-retina.png?ver=e3394"
    />
    <link
      rel="apple-touch-icon"
      sizes="152x152"
      href="touch-icon-ipad-retina.png?ver=e3394"
    />
    <meta
      name="msapplication-TileImage"
      content="touch-w8-mediumtile.png?ver=e3394"
    />
    <meta
      name="msapplication-square70x70logo"
      content="touch-w8-smalltile.png?ver=e3394"
    />
    <meta
      name="msapplication-square150x150logo"
      content="touch-w8-mediumtile.png?ver=e3394"
    />
    <meta
      name="msapplication-wide310x150logo"
      content="touch-w8-widetile.png?ver=e3394"
    />
    <meta
      name="msapplication-square310x310logo"
      content="touch-w8-largetile.png?ver=e3394"
    />
    <link rel="preconnect" href="https://fonts.googleapis.com" />
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
    <link
      href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;500;600;700;900&display=swap"
      rel="stylesheet"
    />
    <title>Frontend Job</title>
  </head>
  <body>
    <noscript>You need to enable JavaScript to run this app.</noscript>
    <div id="root"></div>
  </body>
</html>
