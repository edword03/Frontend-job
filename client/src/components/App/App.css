:root {
  --font-size: 14px;
  --bold: 700;
  --primary-color: #000;
  --default-color: #6B8397;
  --secondary-color: #0070FB;
  --white: #fff;
  --bg-color: #fff;
  --shadow: 0px 3px 15px rgba(0, 0, 0, 0.03);
  --color-b: #000;
  --uppercase: uppercase;
  --hover: #0070FB;
  --text: #9AAAB5;
  --label-c: #49627E;
}

*, *::before, *::after {
  box-sizing: border-box;
}

*, *::before, *:after, *:active, *:focus, *:hover {
  outline: none;
  outline-offset: 0;
}

html {
  font-size: 14px;
}


body {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: 'Montserrat', sans-serif;
  font-size: 14px;
  background-color: #FAFAFA;
}

img {
  max-width: 100%;
  height: auto;
}

h1, h2, h3, h4, h5, h6, p {
  margin: 0;
  padding: 0;
}

a {
  text-decoration: none;
}

ul {
  margin: 0;
  padding: 0;
  list-style: none;
}

figure {
  margin: 0;
  padding: 0;
}
