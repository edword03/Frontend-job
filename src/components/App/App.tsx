import React from 'react';
import { Header, Container } from '@components/index';
import styled, { ThemeContext } from 'styled-components';
import { GlobalStyle, Theme } from '@styles/style.d';
import { FeaturedJobs, SearchJobs } from '@pages/index';

type CurrenPageType = 'searchPage' | 'featuredPage';

type SnippetsType = {
  snippet: {
    requirement: string
    responsibility: string
  }
}

type DataType = {
  snippet: Array<SnippetsType> | never
}

export const App = () => {
  const [currentPage, setCurrenPage] = React.useState<CurrenPageType>('searchPage');
  const [data, setData] = React.useState([])

  const getApi = async(employment: string, experience: string ) => {
    const res = await (await fetch(`https://api.hh.ru/vacancies`)).json()
    const res2 = await (await fetch('https://api.hh.ru/suggests/areas')).json()
    setData(res.items)
  }

  React.useEffect(() => {

    getApi('full', 'between1And3')
    console.log(data);
  }, [])


  const toggleSearchPage = () => {
    setCurrenPage('searchPage');
  }; 

  const toggleFeaturedPage = () => {
    setCurrenPage('featuredPage');
  };

  return (
    <ThemeContext.Provider value={Theme}>
      <GlobalStyle />
      <Header
        currentPage={currentPage}
        setSearchPage={toggleSearchPage}
        setFeaturedPage={toggleFeaturedPage}
      />
      <Container>
        {currentPage === 'searchPage' ? <SearchJobs /> : <FeaturedJobs />}
      </Container>
    </ThemeContext.Provider>
  );
};
