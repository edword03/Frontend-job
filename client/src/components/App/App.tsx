import React from 'react';
import { Header, Container } from '@components/index';
import styled, { ThemeContext } from 'styled-components';
import { GlobalStyle, Theme } from '@styles/style.d';
import { FeaturedJobs, SearchJobs } from '@pages/index';
import { getVacancies } from '@services/api'
import { useQuery, gql } from '@apollo/client';


type CurrenPageType = 'searchPage' | 'featuredPage';


interface ICity {
  name: string;
  id: number;
}

export const JOB_INFO = gql`
  query Job {
    getFetch {
      items {
        name
      }
    }
  }
`

export const App = () => {
  const [currentPage, setCurrenPage] = React.useState<CurrenPageType>('searchPage');
  // const [data, setData] = React.useState({});
  const { loading, error, data, refetch } = useQuery(JOB_INFO)

  // console.log(loading, error, data);

  // data.getFetch.items.map((item: any) => console.log(item))

  const toggleSearchPage = () => {
    setCurrenPage('searchPage');
  };

  const toggleFeaturedPage = () => {
    setCurrenPage('featuredPage');
  };

  return (
      <ThemeContext.Provider value={Theme}>
        <GlobalStyle />
        <Header
          currentPage={currentPage}
          setSearchPage={toggleSearchPage}
          setFeaturedPage={toggleFeaturedPage}
        />
        <Container>{currentPage === 'searchPage' ? <SearchJobs /> : <FeaturedJobs />}</Container>
      </ThemeContext.Provider>
  );
};
