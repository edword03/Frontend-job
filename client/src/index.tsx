import React from 'react';
import ReactDOM from 'react-dom';
import { App } from '@components/App';
import { ApolloClient, gql, ApolloProvider } from '@apollo/client';
import { cache } from './cache';

export const typeDefs = gql`
  type VacancyItem {
    title: String
    description: String
  }

  extend type Query {
    isVisible: Boolean!
    vacancyId: ID!
    currentVacancy: VacancyItem
  }
`;


const client = new ApolloClient({
  uri: process.env.REACT_APP_GRAPHQL_URL,
  cache,
  typeDefs
});


ReactDOM.render(
  <ApolloProvider client={client}>
    <React.StrictMode>
      <App />
    </React.StrictMode>
  </ApolloProvider>,
  document.getElementById('root'),
);
