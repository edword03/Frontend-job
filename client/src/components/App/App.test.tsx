import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import React from 'react';
import { App } from './App';

describe('testing <App />', () => {
  it('should render App', () => {
    const app = render(
      <MockedProvider>
        <App />
      </MockedProvider>,
    );

    expect(app).toMatchSnapshot();
  });
});
