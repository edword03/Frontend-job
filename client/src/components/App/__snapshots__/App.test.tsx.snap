// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <App /> should render App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-gsDKAQ bNwQTa"
      >
        <div
          class="sc-bdvvtL idGvU"
        >
          <div
            class="sc-dkPtRN jwkmEO"
          >
            <a
              href="/"
            >
              <span
                class="sc-hKwDye hQOECq"
              >
                Frontend 
                <span
                  class="sc-eCImPb dieplM"
                >
                  Job
                </span>
              </span>
            </a>
            <nav
              class="sc-jRQBWg ErfWM"
            >
              <a
                class="sc-gKclnd yRrXg"
                href="/jobs"
              >
                Поиск вакансий
              </a>
              <a
                class="sc-gKclnd yRrXg"
                href="/featured-jobs"
              >
                Избранные вакансии
              </a>
            </nav>
          </div>
        </div>
      </header>
      <div
        class="sc-gSQFLo ldXnKf"
      >
        <div>
          <h1
            class="sc-lbhJGD WpgHf"
          >
            Найди работу мечты
          </h1>
          <h2
            class="sc-iNGGcK gfoJkK"
          >
             вакансий доступны прямо сейчас
          </h2>
        </div>
      </div>
      <div
        class="sc-bdvvtL idGvU"
      >
        <section
          class="sc-jeraig dGkmgx"
        >
          <h3
            class="sc-eJwWfJ hQEHvZ"
          >
            Недавние вакансии:
          </h3>
          <div />
          <div
            class="sc-nVkyK jQkhhM"
          >
            <a
              href="/jobs"
            >
              <button
                class="sc-ieecCq gqxRky"
              >
                Посмотреть все
              </button>
            </a>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-gsDKAQ bNwQTa"
    >
      <div
        class="sc-bdvvtL idGvU"
      >
        <div
          class="sc-dkPtRN jwkmEO"
        >
          <a
            href="/"
          >
            <span
              class="sc-hKwDye hQOECq"
            >
              Frontend 
              <span
                class="sc-eCImPb dieplM"
              >
                Job
              </span>
            </span>
          </a>
          <nav
            class="sc-jRQBWg ErfWM"
          >
            <a
              class="sc-gKclnd yRrXg"
              href="/jobs"
            >
              Поиск вакансий
            </a>
            <a
              class="sc-gKclnd yRrXg"
              href="/featured-jobs"
            >
              Избранные вакансии
            </a>
          </nav>
        </div>
      </div>
    </header>
    <div
      class="sc-gSQFLo ldXnKf"
    >
      <div>
        <h1
          class="sc-lbhJGD WpgHf"
        >
          Найди работу мечты
        </h1>
        <h2
          class="sc-iNGGcK gfoJkK"
        >
           вакансий доступны прямо сейчас
        </h2>
      </div>
    </div>
    <div
      class="sc-bdvvtL idGvU"
    >
      <section
        class="sc-jeraig dGkmgx"
      >
        <h3
          class="sc-eJwWfJ hQEHvZ"
        >
          Недавние вакансии:
        </h3>
        <div />
        <div
          class="sc-nVkyK jQkhhM"
        >
          <a
            href="/jobs"
          >
            <button
              class="sc-ieecCq gqxRky"
            >
              Посмотреть все
            </button>
          </a>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
